flang/lib/Lower/RTBuilder.h

```diff
@@ -131,6 +131,13 @@ constexpr TypeBuilderFunc getModel<double &>() {
   };
 }
 template <>
+constexpr TypeBuilderFunc getModel<double *>() {
+  return [](mlir::MLIRContext *context) -> mlir::Type {
+    TypeBuilderFunc f{getModel<double>()};
+    return fir::PointerType::get(f(context));
```

Comment:
fir::PointerType is reserved for Fortran POINTER attribute objects. I think you want fir::ReferenceType to be consistent.

Reply:
done